Only in surf2: config.h
Only in surf2: surf
diff -crB surf/surf.c surf2/surf.c
*** surf/surf.c	2011-05-29 12:31:05.797323000 -0800
--- surf2/surf.c	2011-05-29 12:35:08.897323228 -0800
***************
*** 23,29 ****
  #define LENGTH(x)               (sizeof x / sizeof x[0])
  #define CLEANMASK(mask)         (mask & ~(GDK_MOD2_MASK))
  
! enum { AtomFind, AtomGo, AtomUri, AtomLast };
  
  typedef union Arg Arg;
  union Arg {
--- 23,29 ----
  #define LENGTH(x)               (sizeof x / sizeof x[0])
  #define CLEANMASK(mask)         (mask & ~(GDK_MOD2_MASK))
  
! enum { AtomFind, AtomGo, AtomUri, AtomNav, AtomLast };
  
  typedef union Arg Arg;
  union Arg {
***************
*** 594,599 ****
--- 594,605 ----
  				loaduri(c, &arg);
  				return GDK_FILTER_REMOVE;
  			}
+ 			else if(ev->atom == atoms[AtomNav]) {
+ 				arg.i = atoi(getatom(c, AtomNav));
+ 				navigate(c, &arg);
+ 				return GDK_FILTER_REMOVE;
+ 			}
+ 
  		}
  	}
  	return GDK_FILTER_CONTINUE;
***************
*** 687,692 ****
--- 693,699 ----
  	atoms[AtomFind] = XInternAtom(dpy, "_SURF_FIND", False);
  	atoms[AtomGo] = XInternAtom(dpy, "_SURF_GO", False);
  	atoms[AtomUri] = XInternAtom(dpy, "_SURF_URI", False);
+ 	atoms[AtomNav] = XInternAtom(dpy, "_SURF_NAV", False);
  
  	/* dirs and files */
  	cookiefile = buildpath(cookiefile);
